optaplanner-wb-screens/optaplanner-wb-solver-editor/optaplanner-wb-solver-editor-client/src/test/java/org/optaplanner/workbench/screens/solver/client/editor/TerminationPopupTest.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.optaplanner.workbench.screens.solver.client.editor;
+
+import java.util.ArrayList;
+
+import com.google.gwtmockito.GwtMockitoTestRunner;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.optaplanner.workbench.screens.solver.model.TerminationConfigOption;
+import org.uberfire.commons.data.Pair;
+
+import static org.mockito.Matchers.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(GwtMockitoTestRunner.class)
+public class TerminationPopupTest {
+
+    @Mock
+    private TerminationPopupView view;
+
+    @Mock
+    private TerminationPopupView.TerminationPopupHandler handler;
+
+    @Mock
+    private TerminationTreeItemContent content;
+
+    @Test
+    public void testInit() {
+        TerminationPopup terminationPopup = new TerminationPopup( view );
+        terminationPopup.init( content, new ArrayList<>() );
+        terminationPopup.show();
+
+        verify( view, times( 1 ) ).init( terminationPopup );
+        verify( view, times( 1 ) ).show();
+
+        verify( view, times( 1 ) ).initTerminationOptionList( anyList() );
+        verify( view, times( 1 ) ).showTerminationValueLabel( false );
+        verify( view, times( 1 ) ).showNumericTerminationValueInput( false );
+        verify( view, times( 1 ) ).showTextTerminationValueInput( false );
+        verify( view, times( 1 ) ).showListTerminationValueInput( false );
+    }
+
+    @Test
+    public void testCreateNewTermination() {
```

Comment:
Style suggestion: As of Junit 4 it's no longer needed to use the prefix "test" for method names. For example, this method could be named "createNewTermination()" instead, making it less verbose in the IntellIJ JUnit runner, logs etc.